### ISSUE TITLE
Exception is signal handler error 

### ISSUE THREAD

**Issue by Pitrsek** [horizon-eda/horizon#354]
Hi,
I've created a new unit in pool, saved it. Then I've realized that I made a mistake in file name. I made duplicate with correct file name, and I wanted to delete the original one in the pool. That's when(while deleting from the eda) error window appeared:
![image](https://user-images.githubusercontent.com/16414810/77359101-60aac300-6d4b-11ea-85f1-eaef4d72c1b3.png)

After the error window, the program crashed.   
Windows 10 64bit Build 18363
1.0.0 Aurora
Commit 3f777bd 2020-01-29 19:14:09 +0100 version 1.0.0

**Comment by carrotIndustries** [comment]
Is this error reproducible? This could happen if there was a pool update running.

**Comment by Pitrsek** [comment]
Indeed, when pool update is running, funny things happen. I was able to somehow reproduce it. I was doing the same with entities(copying, renaming, deleting), when I clicked to units tab and double click on unit, I got the same error.  
I also got a new one:
![image](https://user-images.githubusercontent.com/16414810/77363767-96ec4080-6d53-11ea-99f9-df16ccfb9e12.png)
Double click on unit I have just deleted while pool update was in progress(the unit entry was still visible, as pool update has not updated it yet). When I did it again I got a proper message about file not being available

**Comment by carrotIndustries** [comment]
Try the latest build https://ci.appveyor.com/project/carrotIndustries/horizon/builds/31666639 (when it's done, takes a while). To me at least, this completely fixed all brokenness during pool update.

**Comment by Pitrsek** [comment]
I got the same error with a new version when I used it first time. As I'm working with the same parts, maybe pool was corrupted from the previous crash?
![image](https://user-images.githubusercontent.com/16414810/77799490-36bc0e00-7075-11ea-8b57-938a430565c6.png)
I haven't had any issues after pool rebuild/update, but I did not had much time to test it(I've just created one component and proceed to schematic).

**Comment by carrotIndustries** [comment]
You'll need to run the pool update at least once to apply the database fix.
